### PR TITLE
Unify fully qualifed name resolving of imported symbols for all rules

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**

### CODE REVIEW

File: eslint-bridge/src/linting/eslint/rules/helpers/express.ts [hunk 1]
```diff
@@ -108,10 +105,8 @@ export namespace Express {
     n: estree.Node,
   ): boolean {
     if (n.type === 'CallExpression') {
-      const usedMiddleware = getModuleNameOfNode(context, n.callee)?.value;
-      if (usedMiddleware) {
-        return middlewares.includes(String(usedMiddleware));
-      }
+      const fqn = getFullyQualifiedName(context, n);
+      return middlewares.some(middleware => middleware === fqn);
```
Comment (francoismora) [inline eslint-bridge/src/linting/eslint/rules/helpers/express.ts:109]:
```suggestion
      return middlewares.includes(fqn);
```

File: eslint-bridge/src/linting/eslint/rules/helpers/module.ts [hunk 2]
```diff
@@ -278,6 +102,18 @@ export function getFullyQualifiedName(
   let nodeToCheck = reduceToIdentifier(node, fqn);
 
   if (!isIdentifier(nodeToCheck)) {
+    // require chaining, e.g. `require('lib')()` or `require('lib').prop()`
+    if (node.type === 'CallExpression') {
+      const qualifiers: string[] = [];
+      const maybeRequire = reduceTo('CallExpression', node.callee, qualifiers);
+      const module = getModuleNameFromRequire(maybeRequire);
+      if (typeof module?.value === 'string') {
+        const qualifiers: string[] = [];
```
Comment (francoismora) [inline eslint-bridge/src/linting/eslint/rules/helpers/module.ts:111]: `qualifiers` is defined twice. It seems the second one is not necessary.

File: eslint-bridge/src/linting/eslint/rules/os-command.ts [hunk 3]
```diff
@@ -43,54 +39,36 @@ export const rule: Rule.RuleModule = {
...
   if (!command || command.type === 'Literal') {
     return false;
   }
   // for `spawn` and `execFile`, `shell` option must be set to `true`
-  if (isIdentifier(expression, ...SPAWN_EXEC_FILE_FUNCTIONS)) {
+  if (SPAWN_EXEC_FILE_FUNCTIONS.some(fun => fun === method)) {
     return containsShellOption(otherArguments);
   }
-  return isIdentifier(expression, ...EXEC_FUNCTIONS);
+  return EXEC_FUNCTIONS.some(fun => fun === method);
```
Comment (francoismora) [inline eslint-bridge/src/linting/eslint/rules/os-command.ts:71]:
```suggestion
  return EXEC_FUNCTIONS.includes(method);
```

File: eslint-bridge/tests/linting/eslint/rules/comment-based/hashing.js [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
+c.createHash('md4'); // Noncompliant
+c.createHash('md5'); // Noncompliant
+c.createHash('md6'); // Noncompliant
+c.createHash('haval128'); // Noncompliant
+c.createHash('hmacmd5'); // Noncompliant
+c.createHash('dsa'); // Noncompliant
+c.createHash('ripemd'); // Noncompliant
+c.createHash('ripemd128'); // Noncompliant
+c.createHash('ripemd160'); // Noncompliant
+c.createHash('hmacripemd160'); // Noncompliant
```
Comment (francoismora) [inline eslint-bridge/tests/linting/eslint/rules/comment-based/hashing.js:19]: Why did you need to rename the module variable?

**Review by francoismora — COMMENTED** [review]
Looks good and I just have a couple of questions.

File: eslint-bridge/src/linting/eslint/rules/helpers/express.ts (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/src/linting/eslint/rules/helpers/express.ts:109]: That won't compile because `fqn` is of type `string | null` while `middlewares` is of type `string[]`.

File: eslint-bridge/src/linting/eslint/rules/helpers/module.ts (hunk 2 above)
Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/src/linting/eslint/rules/helpers/module.ts:111]: Nice catch, thanks!

File: eslint-bridge/src/linting/eslint/rules/os-command.ts (hunk 3 above)
Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/src/linting/eslint/rules/os-command.ts:71]: 👍

File: eslint-bridge/tests/linting/eslint/rules/comment-based/hashing.js (hunk 4 above)
Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/tests/linting/eslint/rules/comment-based/hashing.js:19]: Because we used to import `crypto` twice: 1) with a `require` call and with an `import` statement. Here is the second occurrence:

https://github.com/SonarSource/SonarJS/blob/dd2b4a2981a50902ee412165d1f1febb51e003e1/eslint-bridge/tests/linting/eslint/rules/comment-based/hashing.js#L27-L28

We ended up in a situation where the symbol `crypto` originated from two sources and polluted the scope, which is not supported by the fully qualified name resolving helper.

https://github.com/SonarSource/SonarJS/blob/dd2b4a2981a50902ee412165d1f1febb51e003e1/eslint-bridge/src/linting/eslint/rules/helpers/module.ts#L286-L288

Therefore, I needed to rename the imported symbol and its occurrences of one of the two sources.

**Review by francoismora — APPROVED** [review]
LGTM
